CMakeLists.txt

```
@@ -145,7 +145,7 @@ endif ()
 
 # testing
 # FIXME (aw): move testing logic into tests/CMakeLists.txt
-if(BUILD_TESTING)
+if((CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME OR ${PROJECT_NAME}_BUILD_TESTING) AND BUILD_TESTING)
```

```suggestion
if( (CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_TESTING)  OR ${PROJECT_NAME}_BUILD_TESTING )
```

Either `BUILD_TESTING` is set globally (therefore filter it) or `${PROJECT_NAME}_BUILD_TESTING` is used.